### ISSUE TITLE
pip broken

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#557]
syntax:
~~~
conda install pip
~~~
output:
~~~
Fetching package metadata: ....
Solving package specifications: ................
# All requested packages already installed.
# packages in environment at /Users/julien/miniconda:
#
pip                       7.1.2                    py27_0  
~~~

syntax:
~~~
pip
~~~

output:
~~~
Traceback (most recent call last):
  File "/Users/julien/miniconda/bin/pip", line 4, in <module>
    from pip import main
  File "/Users/julien/miniconda/lib/python2.7/site-packages/pip/__init__.py", line 12, in <module>
    from pip.exceptions import InstallationError, CommandError, PipError
ImportError: No module named exceptions
~~~

Station: OSX 10.8.5

**Comment by jcohenadad** [comment]
Benjamin I assigned you but I'm also looking into this (everyone should look into this urgent matter)

**Comment by jcohenadad** [comment]
ok, one way to overcome the problem is to install pip with easy_install

**Comment by benjamindeleener** [comment]
I don't have this issue. Did you try to reinstall pip? (uninstall and install again) Or to upgrade pip using conda? easy_install is a good solution too but conda should install pip normally.

**Comment by jcohenadad** [comment]
works on 10.10, 10.9 but not on 10.8.
easy_install works on 10.10, 10.9 and 10.8.  So i'll make the change for easy_install.

**Comment by jcohenadad** [comment]
and yes, i did uninstall pip, reinstalled, etc.

**Comment by jcohenadad** [comment]
N.B. I used :``commands.getstatusoutput`` instead of ``runProcess`` because if runProcess fails, it automatically exists the program.

**Comment by jcohenadad** [comment]
Update: same issue with 10.10.4, after removing mac version of pip and installing it within miniconda:

~~~
pip
Traceback (most recent call last):
  File "/Users/julien/miniconda/bin/pip", line 9, in <module>
    load_entry_point('pip==7.1.2', 'console_scripts', 'pip')()
  File "/Users/julien/miniconda/lib/python2.7/site-packages/setuptools-18.3.2-py2.7.egg/pkg_resources/__init__.py", line 558, in load_entry_point
    
  File "/Users/julien/miniconda/lib/python2.7/site-packages/setuptools-18.3.2-py2.7.egg/pkg_resources/__init__.py", line 2682, in load_entry_point
    
  File "/Users/julien/miniconda/lib/python2.7/site-packages/setuptools-18.3.2-py2.7.egg/pkg_resources/__init__.py", line 2355, in load
    
  File "/Users/julien/miniconda/lib/python2.7/site-packages/setuptools-18.3.2-py2.7.egg/pkg_resources/__init__.py", line 2361, in resolve
    
ImportError: No module named pip
~~~

**Comment by jcohenadad** [comment]
update:  when installing pip with easy_install from miniconda, it also breaks.

**Comment by jcohenadad** [comment]
ok, more investigations (on 10.10.4 with miniconda):
- removed all pip.
- open new terminal
- conda install pip
- open new terminal
- pip --> worked
- conda uninstall pip
- open new terminal
- easy_install pip
- open new terminal 
- pip --> does not work (same error as above)

**Comment by jcohenadad** [comment]
on another station, with clean installation of miniconda, i get the following:
~~~
easy_install pip
error: can't create or remove files in install directory

The following error occurred while trying to add or remove files in the
installation directory:

    [Errno 13] Permission denied: '/Library/Python/2.7/site-packages/test-easy-install-3962.pth'

The installation directory you specified (via --install-dir, --prefix, or
the distutils default setting) was:

    /Library/Python/2.7/site-packages/

Perhaps your account does not have write access to this directory?  If the
installation directory is a system-owned directory, you may need to sign in
as the administrator or "root" account.  If you do not have administrative
access to this machine, you may wish to choose a different installation
directory, preferably one that is listed in your PYTHONPATH environment
variable.

For information on other options, you may wish to consult the
documentation at:

  https://pythonhosted.org/setuptools/easy_install.html

Please make the appropriate changes for your system and try again.
~~~

**Comment by benjamindeleener** [comment]
Can you try
~~~
which easy_install
~~~
Mine is from miniconda

**Comment by jcohenadad** [comment]
strange strange. Now the following works:
- conda uninstall pip
- open new terminal
- pip --> ``-bash: pip: command not found``
- which easy_install --> ``/Users/sct/miniconda/bin/easy_install``
- easy_install pip -->
~~~
Searching for pip
Reading https://pypi.python.org/simple/pip/
Best match: pip 7.1.2
Downloading https://pypi.python.org/packages/source/p/pip/pip-7.1.2.tar.gz#md5=3823d2343d9f3aaab21cf9c917710196
Processing pip-7.1.2.tar.gz
Writing /var/folders/yr/xqk92yt129xg5nj1w343mg9w0000gp/T/easy_install-uLpeJi/pip-7.1.2/setup.cfg
Running pip-7.1.2/setup.py -q bdist_egg --dist-dir /var/folders/yr/xqk92yt129xg5nj1w343mg9w0000gp/T/easy_install-uLpeJi/pip-7.1.2/egg-dist-tmp-BI9xJY
warning: no previously-included files found matching '.coveragerc'
warning: no previously-included files found matching '.mailmap'
warning: no previously-included files found matching '.travis.yml'
warning: no previously-included files found matching 'pip/_vendor/Makefile'
warning: no previously-included files found matching 'tox.ini'
warning: no previously-included files found matching 'dev-requirements.txt'
no previously-included directories found matching '.travis'
no previously-included directories found matching 'docs/_build'
no previously-included directories found matching 'contrib'
no previously-included directories found matching 'tasks'
no previously-included directories found matching 'tests'
creating /Users/sct/miniconda/lib/python2.7/site-packages/pip-7.1.2-py2.7.egg
Extracting pip-7.1.2-py2.7.egg to /Users/sct/miniconda/lib/python2.7/site-packages
Adding pip 7.1.2 to easy-install.pth file
Installing pip script to /Users/sct/miniconda/bin
Installing pip2.7 script to /Users/sct/miniconda/bin
Installing pip2 script to /Users/sct/miniconda/bin

Installed /Users/sct/miniconda/lib/python2.7/site-packages/pip-7.1.2-py2.7.egg
Processing dependencies for pip
Finished processing dependencies for pip
~~~
- pip --> worked!

Usage:   
  pip <command> [options]

Commands:
  install                     Install packages.
  uninstall                   Uninstall packages.
  freeze                      Output installed packages in requirements format.
  list                        List installed packages.
  show                        Show information about installed packages.
  search                      Search PyPI for packages.
  wheel                       Build wheels from your requirements.
  help                        Show help for commands.

General Options:
  -h, --help                  Show help.
  --isolated                  Run pip in an isolated mode, ignoring environment variables and user
                              configuration.
  -v, --verbose               Give more output. Option is additive, and can be used up to 3 times.
  -V, --version               Show version and exit.
  -q, --quiet                 Give less output.
  --log <path>                Path to a verbose appending log.
  --proxy <proxy>             Specify a proxy in the form [user:passwd@]proxy.server:port.
  --retries <retries>         Maximum number of retries each connection should attempt (default 5
                              times).
  --timeout <sec>             Set the socket timeout (default 15 seconds).
  --exists-action <action>    Default action when a path already exists: (s)witch, (i)gnore,
                              (w)ipe, (b)ackup.
  --trusted-host <hostname>   Mark this host as trusted, even though it does not have valid or any
                              HTTPS.
  --cert <path>               Path to alternate CA bundle.
  --client-cert <path>        Path to SSL client certificate, a single file containing the private
                              key and the certificate in PEM format.
  --cache-dir <dir>           Store the cache data in <dir>.
  --no-cache-dir              Disable the cache.
  --disable-pip-version-check
                              Don't periodically check PyPI to determine whether a new version of
                              pip is available for download. Implied with --no-index.

**Comment by jcohenadad** [comment]
OK, I think the problem is related to the call of easy_install:
~~~
julien-macbook:~ $ pip
-bash: pip: command not found
julien-macbook:~ $ 
julien-macbook:~ $ 
julien-macbook:~ $ which python
/Users/julien/miniconda/bin/python
julien-macbook:~ $ which conda
/Users/julien/miniconda/bin/conda
julien-macbook:~ $ pip
-bash: pip: command not found
julien-macbook:~ $ which easy_install
/usr/bin/easy_install
julien-macbook:~ $ easy_install pip
error: can't create or remove files in install directory

The following error occurred while trying to add or remove files in the
installation directory:

    [Errno 13] Permission denied: '/Library/Python/2.7/site-packages/test-easy-install-5826.pth'

The installation directory you specified (via --install-dir, --prefix, or
the distutils default setting) was:

    /Library/Python/2.7/site-packages/

Perhaps your account does not have write access to this directory?  If the
installation directory is a system-owned directory, you may need to sign in
as the administrator or "root" account.  If you do not have administrative
access to this machine, you may wish to choose a different installation
directory, preferably one that is listed in your PYTHONPATH environment
variable.

For information on other options, you may wish to consult the
documentation at:

  https://pythonhosted.org/setuptools/easy_install.html

Please make the appropriate changes for your system and try again.

julien-macbook:~ $ pip
-bash: pip: command not found
julien-macbook:~ $ conda install pip
Fetching package metadata: ....
Solving package specifications: .
Package plan for installation in environment /Users/julien/miniconda:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    python-2.7.10              |                1         9.9 MB
    setuptools-18.3.2          |           py27_0         342 KB
    wheel-0.26.0               |           py27_1          76 KB
    conda-3.18.1               |           py27_0         172 KB
    pip-7.1.2                  |           py27_0         1.4 MB
    ------------------------------------------------------------
                                           Total:        11.8 MB

The following NEW packages will be INSTALLED:

    pip:        7.1.2-py27_0 
    setuptools: 18.3.2-py27_0
    wheel:      0.26.0-py27_1

The following packages will be UPDATED:

    conda:      3.16.0-py27_0 --> 3.18.1-py27_0
    python:     2.7.10-0      --> 2.7.10-1     

Proceed ([y]/n)? 

Fetching packages ...
python-2.7.10- 100% |####################################################| Time: 0:00:00  19.68 MB/s
setuptools-18. 100% |####################################################| Time: 0:00:00   1.67 MB/s
wheel-0.26.0-p 100% |####################################################| Time: 0:00:00 973.62 kB/s
conda-3.18.1-p 100% |####################################################| Time: 0:00:00   1.13 MB/s
pip-7.1.2-py27 100% |####################################################| Time: 0:00:00   3.33 MB/s
Extracting packages ...
[      COMPLETE      ]|#######################################################################| 100%
Unlinking packages ...
[      COMPLETE      ]|#######################################################################| 100%
Linking packages ...
[      COMPLETE      ]|#######################################################################| 100%
julien-macbook:~ $ pip

Usage:   
  pip <command> [options]

Commands:
  install                     Install packages.
  uninstall                   Uninstall packages.
  freeze                      Output installed packages in requirements format.
  list                        List installed packages.
  show                        Show information about installed packages.
  search                      Search PyPI for packages.
  wheel                       Build wheels from your requirements.
  help                        Show help for commands.

General Options:
  -h, --help                  Show help.
  --isolated                  Run pip in an isolated mode, ignoring environment variables and user
                              configuration.
  -v, --verbose               Give more output. Option is additive, and can be used up to 3 times.
  -V, --version               Show version and exit.
  -q, --quiet                 Give less output.
  --log <path>                Path to a verbose appending log.
  --proxy <proxy>             Specify a proxy in the form [user:passwd@]proxy.server:port.
  --retries <retries>         Maximum number of retries each connection should attempt (default 5
                              times).
  --timeout <sec>             Set the socket timeout (default 15 seconds).
  --exists-action <action>    Default action when a path already exists: (s)witch, (i)gnore,
                              (w)ipe, (b)ackup.
  --trusted-host <hostname>   Mark this host as trusted, even though it does not have valid or any
                              HTTPS.
  --cert <path>               Path to alternate CA bundle.
  --client-cert <path>        Path to SSL client certificate, a single file containing the private
                              key and the certificate in PEM format.
  --cache-dir <dir>           Store the cache data in <dir>.
  --no-cache-dir              Disable the cache.
  --disable-pip-version-check
                              Don't periodically check PyPI to determine whether a new version of
                              pip is available for download. Implied with --no-index.

~~~

**Comment by jcohenadad** [comment]
ok, things seem to work now with conda install pip.
fixed in commit: 3216dd30265820f79c94a711ec2b9e833673a3d5
